### PR TITLE
feat(UI): Display Licenses from Scanner findings in ISR attachments

### LINKED ISSUE #1491: (UI): Display License Info from Fossology scanner findings via ISR attachment
[//]: # (Copyright Siemens AG, 2021. Part of the SW360 Portal Project)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

### Description
Please refer issue #1390  and PR #1391  to get more idea on this issue.
This feature is based on above issue and enhancement to the above PR.

- Once `Initial Scan Report` (SPDX-RDF file) is uploaded to release, the Release status will change to `Scan Available`.

At Project Level: In `License Clearing` tab

- For all the release with `Scan Available` status, an `info` icon should be displayed beside the release name.
- Clicking on the `info` icon should display the pop-up with list of License found by fossology scanner, and the ISR file name.

At Release Level: In `Clearing Details` tab

- All the available `ISR` should be listed in `SPDX Attachments` table (if and only if `CLI` is not available).
- None of the `ISR` should be listed in `SPDX Attachments` table, if any `CLI` is available in Release.
- On clicking of `Show License Info` button, all the license in `ISR` file should be displayed in `Result` column of the `SPDX Attachments` table.
- Each License should have `info` icon, clicking on this icon should display a pop-up with all the associated source files with that License.

### PR DESCRIPTION
**Pull request by akapti**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
> * Did you add or update any new dependencies that are required for your change?

Issue: #1491

Closes #1491 

### Screenshot:
**Project UI:**
image
image

**Release UI:**
image
image
image

### Checklist
Must:
- [x] All related issues are referenced in commit messages and in PR

Signed-off-by: Abdul Kapti <abdul.kapti@siemens-healthineers.com>

### CODE REVIEW

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java [hunk 1]
```diff
@@ -573,35 +572,54 @@ private void loadSpdxLicenseInfo(ResourceRequest request, ResourceResponse respo
...
                     .collect(Collectors.toList());
-            if (attachmentName.endsWith(".rdf")) {
-                concludedLicenseIds = licenseInfoResult.stream()
-                        .flatMap(singleResult -> singleResult.getLicenseInfo().getConcludedLicenseIds().stream())
-                        .collect(Collectors.toSet());
-                otherLicenseNames = licenseWithTexts.stream().map(LicenseNameWithText::getLicenseName).collect(Collectors.toSet());
+
+            if (attachmentName.endsWith(PortalConstants.RDF_FILE_EXTENSION)) {
+                if (AttachmentType.INITIAL_SCAN_REPORT.equals(attachmentType)) {
+                    totalFileCount = licenseWithTexts.stream().map(LicenseNameWithText::getSourceFiles).filter(src -> src != null).mapToInt(Set::size).sum();
```
Comment (smrutis1) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java:600]: `Objects::nonNull` will look good may be

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java [hunk 2]
```diff
@@ -2746,6 +2749,92 @@ private void addLicenseToLinkedReleases(ResourceRequest request, ResourceRespons
...
+            attachments = attachments.stream().filter(isISR).collect(Collectors.toSet());
+            if (attachments.size() == 1) {
+                Attachment attachment = attachments.iterator().next();
+                attachmentType = attachment.getAttachmentType();
+                attachmentContentId = attachment.getAttachmentContentId();
+                attachmentName = attachment.getFilename();
+                List<LicenseInfoParsingResult> licenseInfoResult = licenseInfoClient.getLicenseInfoForAttachment(release,
+                        attachmentContentId, true, user);
+                List<LicenseNameWithText> licenseWithTexts = licenseInfoResult.stream()
+                        .flatMap(result -> result.getLicenseInfo().getLicenseNamesWithTexts().stream())
```
Comment (smrutis1) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java:2780]: May be a null check on `getLicenseInfo()` and `getLicenseNamesWithTexts()`

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java [hunk 3]
```diff
@@ -2746,6 +2749,92 @@ private void addLicenseToLinkedReleases(ResourceRequest request, ResourceRespons
...
+                        .filter(license -> !license.getLicenseName().equalsIgnoreCase(SW360Constants.LICENSE_NAME_UNKNOWN)
+                                && !license.getLicenseName().equalsIgnoreCase(SW360Constants.NA)
+                                && !license.getLicenseName().equalsIgnoreCase(SW360Constants.NO_ASSERTION)) // exclude unknown, n/a and noassertion
+                        .collect(Collectors.toList());
+                if (attachmentName.endsWith(PortalConstants.RDF_FILE_EXTENSION)) {
+                    totalFileCount = licenseWithTexts.stream().map(LicenseNameWithText::getSourceFiles).filter(src -> src != null).mapToInt(Set::size).sum();
```
Comment (smrutis1) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java:2786]: `Objects::nonNull`

File: frontend/sw360-portlet/src/main/resources/META-INF/resources/html/components/includes/releases/clearingDetails.jspf [hunk 4]
```diff
@@ -45,13 +45,15 @@
         <core_rt:forEach items="${spdxAttachments}" var="spdxAttachment">
             <tr id="${spdxAttachment.attachmentContentId}">
                 <td>
-                    <sw360:out value="${spdxAttachment.filename}"/>
+                    <div <core_rt:if test="${spdxAttachment.attachmentType eq 'INITIAL_SCAN_REPORT'}">class="actions d-inline" </core_rt:if> >
+                        <sw360:out value="${spdxAttachment.filename}"/><core_rt:if test="${spdxAttachment.attachmentType eq 'INITIAL_SCAN_REPORT'}"><svg class="cursor lexicon-icon text-danger ml-2"><title><liferay-ui:message key='this.is.only.an.initial.scanner.isr.finding.of.the.contained.licenses.in.the.uploaded.source.file' /><br><liferay-ui:message key='the.final.license.list.may.differ.based.on.the.conclusions.made.by.the.clearing.expert.and.is.made.available.as.component.license.information.cli' /></title><use href='/o/org.eclipse.sw360.liferay-theme/images/clay/icons.svg#warning' /></svg></core_rt:if>
```
Comment (smrutis1) [inline frontend/sw360-portlet/src/main/resources/META-INF/resources/html/components/includes/releases/clearingDetails.jspf:49]: Try to break the line for readability

File: frontend/sw360-portlet/src/main/resources/META-INF/resources/html/components/includes/releases/clearingDetails.jspf [hunk 5]
```diff
@@ -231,16 +234,42 @@
...
 
+        $("table#spdxAttachments").on("click", "svg.isr", function(event) {
+            let licenseName = $(event.currentTarget).data().licenseId,
+                attId = $(event.currentTarget).closest('tr').attr('id'),
+                attName = $(event.currentTarget).closest('tr').find('td:first div')[0].innerText;
+            displayLicenseToSrcMapping(attId, attName, licenseName);
+        });
+
+        function displayLicenseToSrcMapping(attId, attName, licenseName) {
+        	let data,
```
Comment (smrutis1) [inline frontend/sw360-portlet/src/main/resources/META-INF/resources/html/components/includes/releases/clearingDetails.jspf:245]: Indentation

**Review by smrutis1 — CHANGES_REQUESTED** [review]
Some minor changes requested

**Comment by ag4ums** [comment]
Multiple ISR in a release is not process correctly... may consider the last report uploaded(timestamp from filename of spdx), Or warning message like multiple ISR found.....
image

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java (hunk 1 above)
Comment (akapti) [reply frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java]: Added `Objects::nonNull`

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java (hunk 2 above)
Comment (akapti) [reply frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java:2784]: added bull check in `stream().filter()`

File: frontend/sw360-portlet/src/main/resources/META-INF/resources/html/components/includes/releases/clearingDetails.jspf (hunk 4 above)
Comment (akapti) [reply frontend/sw360-portlet/src/main/resources/META-INF/resources/html/components/includes/releases/clearingDetails.jspf]: Done.

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java (hunk 3 above)
Comment (akapti) [reply frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java]: Added `Objects::nonNull`

File: frontend/sw360-portlet/src/main/resources/META-INF/resources/html/components/includes/releases/clearingDetails.jspf (hunk 5 above)
Comment (akapti) [reply frontend/sw360-portlet/src/main/resources/META-INF/resources/html/components/includes/releases/clearingDetails.jspf]: fixed.

**Comment by akapti** [comment]
> Multiple ISR in a release is not process correctly... may consider the last report uploaded(timestamp from filename of spdx), Or warning message like multiple ISR found..... 

The issue has been fixed.
Now proper message is being displayed incase of Multiple ISR or No ISR are found in release.

**Review by ag4ums — APPROVED** [review]
working as expected

**Review by smrutis1 — APPROVED** [review]
Code looks good.

**Pull request merged**